client/src/pages/Events/Events.tsx

```diff
@@ -5,27 +5,96 @@ import { useAuth } from "../../contexts/AuthContext";
 import { toastNotify } from "../../utils/ui";
+import { processVolunteerCount } from "../ViewEvent/VolunteersTable";
+import { useEffect, useState } from "react";
+import { ProcessedScheduledSlot } from "../../types"; // Ensure this path is correct
 
 const newEventLink =
   "https://airtable.com/shrETAYONKTJMVTnZ?prefill_Supplier=Rap+4+Bronx&prefill_Start+Time=01/01/2023+09:00am&prefill_End+Time=01/01/2023+01:00pm&prefill_First+Driving+Slot+Start+Time=01/01/2023+10:30am&prefill_How+long+should+each+Driver+Time+Slot+be?=0:15&prefill_Max+Count+of+Drivers+Per+Slot=30&prefill_How+long+should+the+Logistics+slot+be?=1:30&prefill_Maximum+number+of+drivers+needed+for+this+event+(usually+30)?=30&prefill_Max+Count+of+Distributors+Per+Slot=30";
 
+//made to avoid implicit any type errors
+interface Event {
+  id: string;
+  dateDisplay: string;
+  time: string;
+  mainLocation: string;
+  numTotalParticipants: number;
+  numDrivers: number;
+  numPackers: number;
+  scheduledSlots: string[];
+}
+
+interface EventWithVolunteersData extends Event {
+  volunteers: ProcessedScheduledSlot[];
+}
+
 export function Events() {
   const { token, setToken } = useAuth();
   if (!token) {
     return <Navigate to="/" />;
   }
   const futureEventsQuery = useFutureEvents();
+  const [eventsData, setEventsData] = useState<EventWithVolunteersData[]>([]);
+  const [loading, setLoading] = useState(true);
+  const [displayCount, setDisplayCount] = useState(8);
+
+  useEffect(() => {
+    const fetchVolunteersForEvent = async (event: Event, token: string) => {
+      const response = await fetch(
+        `/api/volunteers/?scheduledSlotsIds=${event.scheduledSlots.join(",")}`,
+        {
+          headers: { Authorization: `Bearer ${token}` },
+        }
+      );
+      if (!response.ok) {
+        const data = await response.json();
+        throw new Error(data.message);
+      }
+      return response.json() as Promise<ProcessedScheduledSlot[]>;
+    };
 
-  if (
-    futureEventsQuery.status === "loading" ||
-    futureEventsQuery.status === "idle"
-  ) {
+    const fetchVolunteersForAllEvents = async () => {
+      if (futureEventsQuery.status === "success" && futureEventsQuery.data) {
+        try {
+          const eventsWithVolunteers: EventWithVolunteersData[] =
+            await Promise.all(
+              futureEventsQuery.data
+                .slice(0, displayCount)
+                .map(async (event) => {
+                  const volunteers = await fetchVolunteersForEvent(
+                    event,
+                    token
+                  );
+                  return {
+                    ...event,
+                    volunteers,
+                  };
+                })
+            );
+          setEventsData(eventsWithVolunteers);
+          setLoading(false);
+        } catch (error) {
+          console.error(error);
+          setLoading(false);
+        }
+      }
+    };
+
+    fetchVolunteersForAllEvents();
```

Comment:
Echoing what @mattsahn said about avoiding fetching individually. Compute if at all possible or at least batch. Also, this is making a brand new call on every single data change, which means you can easily cascade hundred of queries if events are updating. Also, this seems to be composing a status out of multiple existing queries under react-query. If we must, we should at least use their composition patterns (dependent queries)